server/models/video/video-query-builder.ts

```diff
@@ -239,16 +241,65 @@ function buildListQuery (model: typeof Model, options: BuildVideosQueryOptions)
     }
   }
 
-  // We don't exclude results in this if so if we do a count we don't need to add this complex clauses
+  // We don't exclude results in this so if we do a count we don't need to add this complex clause
   if (options.trendingDays && options.isCount !== true) {
     const viewsGteDate = new Date(new Date().getTime() - (24 * 3600 * 1000) * options.trendingDays)
 
     joins.push('LEFT JOIN "videoView" ON "video"."id" = "videoView"."videoId" AND "videoView"."startDate" >= :viewsGteDate')
     replacements.viewsGteDate = viewsGteDate
 
-    attributes.push('COALESCE(SUM("videoView"."views"), 0) AS "videoViewsSum"')
+    attributes.push('COALESCE(SUM("videoView"."views"), 0) AS "score"')
 
     group = 'GROUP BY "video"."id"'
+  } else if (options.hot && options.isCount !== true) {
+    /**
+     * "Hotness" is a measure based on absolute view/comment/like/dislike numbers,
+     * with fixed weights only applied to their log values.
+     *
+     * This algorithm gives little chance for an old video to have a good score,
+     * for which recent spikes in interactions could be a sign of "hotness" and
+     * justify a better score. However there are multiple ways to achieve that
+     * goal, which is left for later. Yes, this is a TODO :)
+     *
+     * note: weights and base score are in number of half-days.
+     * see https://github.com/reddit-archive/reddit/blob/master/r2/r2/lib/db/_sorts.pyx#L47-L58
+     */
+    const weights = {
+      like: 3,
+      dislike: 3,
+      view: 1 / 12,
+      comment: 2 // a comment takes more time than a like to do, but can be done multiple times
+    }
+
+    cte.push( // TODO: exclude blocklisted comments
+      '"totalCommentsWithoutVideoAuthor" AS (' +
```

Comment:
I don't think it will scale: you're building a temp' table for every videos and comments in the DB.

Reply:
You're right. Maybe as a computed field on video?

Reply:
No sorry, we can't store and manage another field just for 1 specific sort of 1 query

Reply:
Ok, I'll simplify then.

Reply:
Should be better.